h/migrations/versions/59f42b6a0cd9_add_group_creators_as_owners.py

```diff
@@ -0,0 +1,51 @@
+"""Add existing group creators as owners."""
+
+from alembic import op
+from sqlalchemy import Column, ForeignKey, Integer, MetaData, Table, select, update
+from sqlalchemy.dialects.postgresql import JSONB
+
+metadata = MetaData()
+
+
+user_table = Table("user", metadata, Column("id", Integer, primary_key=True))
+
+
+group_table = Table(
+    "group",
+    metadata,
+    Column("id", Integer, primary_key=True),
+    Column("creator_id", Integer, ForeignKey("user.id")),
+)
+
+
+user_group_table = Table(
+    "user_group",
+    metadata,
+    Column("id", Integer, primary_key=True),
+    Column("user_id", Integer, ForeignKey("user.id")),
+    Column("group_id", Integer, ForeignKey("group.id")),
+    Column("roles", JSONB),
+)
+
+
+revision = "59f42b6a0cd9"
+down_revision = "1975edef158d"
+
+
+def upgrade():
+    op.execute(
+        update(user_group_table)
+        .where(
+            user_group_table.c.id.in_(
+                select(user_group_table.c.id)
+                .where(user_group_table.c.group_id == group_table.c.id)
+                .where(user_group_table.c.user_id == user_table.c.id)
+                .where(user_table.c.id == group_table.c.creator_id)
+            )
+        )
+        .values(roles=["owner"])
+    )
```

Comment:
Not all existing groups will have creators. There are three cases:

1. The user account of the group's creator has been deleted. In this case `"group".creator_id` is `NULL`. There's nothing to do in this case.
2. The creator is no longer a member of the group. In this case `"group".creator_id` will still correspond to an existing `"user".id` but there will be no `user_group` row with a matching `group_id` and `user_id`. Again there's nothing to do in this case.
3. The creator is still a member of the group. In this case we find the right `user_group` row and update `roles` to `"owner"`.

And of course we need to _not_ update `user_group` roles for non-creator memberships.

I'm hoping I can get away with updating all the rows at once as in the SQL query above. In the production DB there are 191,571 rows matching:

```sql
SELECT count(*) FROM "group", "user", user_group WHERE "group".id = user_group.group_id AND "user".id = user_group.user_id AND "group".creator_id = "user".id;
```

Reply:
I'm hoping I can get away with updating all the rows at once as in the SQL query above. In the production DB there are 191,571 rows matching:

I reckon it would be fine but we should run it early in the morning just in case.